lnrpc/routerrpc/config.go

```diff
@@ -10,9 +10,18 @@ import (
 // The fields with struct tags are meant to be parsed as normal configuration
 // options, while if able to be populated, the latter fields MUST also be
 // specified.
+//
+//nolint:lll
 type Config struct {
 	RoutingConfig
 
+	// UseStatusInitiated is a boolean that indicates whether the router
+	// should use the new status code `Payment_INITIATED`.
+	//
+	// TODO(yy): remove this config after the new status code is fully
+	// deployed to the network(v0.20.0).
+	UseStatusInitiated bool `long:"usestatusinitiated" description:"If true, the router will send Payment_INITIATED for new payments, otherwise Payment_In_FLIGHT will be sent for compatibility concerns."`
```

Comment:
yeah think it'd give people enough time to upgrade? The other approach is to let users specify an RPC version in `SendPayment` request, but that would end up being messy. My concern is, if people do not use this new flag in the coming versions, we'd still have this compatibility issue.

Reply:
> yeah think it'd give people enough time to upgrade? The other approach is to let users specify an RPC version in SendPayment request, but that would end up being messy. 

Makes sense! Agree that putting it on the RPC is going to be ugly, we use the payment state in too many places (eg, would also need it in `ListPayments`, for example). 

> My concern is, if people do not use this new flag in the coming versions, we'd still have this compatibility issue.

Yah this is tough. Could force it more by making this flag `DisableStatusInitiated` and then people have to set it to prevent the breaking change, but there really is no easy path :(